tsa-core/src/main/kotlin/org/usvm/machine/interpreter/TvmTransactionInterpreter.kt

```diff
@@ -242,33 +251,57 @@ class TvmTransactionInterpreter(
                     restActions(currentMessageHandlingState)
                     return
                 }
-            val mode = head.outMessage.sendMessageMode
+            val mode = head.sendMessageMode
             val sendRemainingValue = mode.hasBitSet(MessageMode.SEND_REMAINING_VALUE_BIT)
             val sendRemainingBalance = mode.hasBitSet(MessageMode.SEND_REMAINING_BALANCE_BIT)
             val sendFwdFeesSeparately = mode.hasBitSet(MessageMode.SEND_FEES_SEPARATELY)
-            val messageValue = head.outMessage.content.commonMessageInfo.msgValue
-            ctx.handleSingleMessage(
-                scope = this@handleMessagesImpl,
-                sendRemainingValue = sendRemainingValue,
-                sendRemainingBalance = sendRemainingBalance,
-                sendFwdFeesSeparately = sendFwdFeesSeparately,
-                computeFees = zeroValue,
-                initMsgValue = messageValue,
-                currentState = currentMessageHandlingState,
-                currentMessage = head,
-            ) { newCurrentState ->
-                handleMessagesImpl(tail, newCurrentState, restActions)
+            val sendIgnoreErrors =
+                mode
+                    .hasBitSet(MessageMode.SEND_IGNORE_ERRORS)
+                    .asIntValue()
+                    .intValueOrNull
+                    ?.let { it != 0 }
+                    ?: error("Only concrete mode is supported")
```

Comment:
Is it? I am not against it, it just seems that a lot of things could be simplified if we supported only concrete mode